src/ghost_borrow_mut.rs

```diff
@@ -247,6 +246,63 @@ generate_public_instance!(a, b, c, d, e, f, g, h, i, j, k, l ; T0, T1, T2, T3, T
 //  Implementation
 //
 
+//  This trait is a pure implementation detail, which only needs to be public because it must be part of the signature
+//  of public functions/traits.
+mod sealed {
+
+#[derive(Clone, Copy, Debug)]
+pub enum Span {
+    Single(*const u8),
+    Slice(*const u8, *const u8),
+}
+
+#[doc(hidden)]
+pub trait GetSpan {
```

Comment:
I really don't like that you're specializing over slices specifically. I think that every object's span should be checked to be non overlapping, regardless if it's a slice.

The only reason to treat slices differently is because `as_slice_of_cells` is specific to slices. However, issue #23 's problem is that in general, comparing pointers of objects of different types isn't enough to determine that they're non overlapping.

In fact, I think that slices shouldn't be treated differently, and that `Span::Single` shouldn't exist.

Reply:
I've been thinking about it, and it's actually... a bad idea.

Specifically, the issue is that it's possible to have a `GhostCell` to a `struct` whose last item is a slice: it won't take the slice specialization, yet will have all the same issues.